### DISCUSSION TITLE
Use of TFT_eSPI in Platformio

### DISCUSSION THREAD

**Discussion by videobuff** [Bodmer/TFT_eSPI#967] [Q&A - General]
I have setup platformio and i am able to compile and upload to my ESP32. 
However, if I choose the option to add the TFT_eSPI  library from platformio to my project after I added it to platformio, i get errors.
I cannot find, where user-select.h is located ? And there is no library in the lib directory under my project. So i get these errors: 
`src/main.cpp:118:25: error: 'class TFT_eSPI' has no member named 'getTouch'`

If I add the whole TFT_eSPI library to my project, it works.
Anyone has a soluton for this ?
Erik

![Schermafbeelding 2021-01-30 om 14 46 39](https://user-images.githubusercontent.com/3132953/106358049-18b0e100-630a-11eb-9a3b-e6052a7aad11.jpg)

**Comment by GioCC** [comment]
PlatformIO-supplied libraried are copied under ".\.pio\libdeps".

**Comment by djp456** [comment]
PlatformIO uses a "normal" gcc compiler, which doesn't have some of the training wheels of the Arduino compiler. In particular, all functions must be defined before getting to "main" (or setup() and loop() in Arduino-speak). I'm fairly certain gettouch() is defined below loop() in your source file. If so, cut and paste the code for setup() and loop() code so those are the last two things in your source file, and it should compile fine.
